### PR TITLE
Small changes to where penalties are applied

### PR DESCRIPTION
**Pull request by tobiasbecher**
Moved the penalties from the objectives to the objective function. Also added an option for objective constraint.

### CODE REVIEW

File: optimization/+DoseConstraints/matRad_ObjectiveConstraint.m [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+            if nargin == 1 && isstruct(objective)
+                initFromStruct = true;
+                inputStruct = objective;
+            else
+                initFromStruct = false;
+                inputStruct = [];
+            end
+
+            obj@DoseConstraints.matRad_DoseConstraint(inputStruct);
```
Comment (amitantony) [inline optimization/+DoseConstraints/matRad_ObjectiveConstraint.m:48]: for readability i would change this 'obj' named instance to something else, 'doseConstraint'

File: optimization/+DoseConstraints/matRad_ObjectiveConstraint.m [hunk 2]
```diff
@@ -0,0 +1,111 @@
+classdef matRad_ObjectiveConstraint < DoseConstraints.matRad_DoseConstraint
```
Comment (wahln) [inline optimization/+DoseConstraints/matRad_ObjectiveConstraint.m:1]: I find the name a little ambiguous. Could we maybe switch to matRad_DoseConstraintFromObjective?

File: optimization/+DoseConstraints/matRad_ObjectiveConstraint.m [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    properties (Constant)
+        name = 'Objective Constraint';
+        parameterTypes = {'objFunc'};
+        parameterNames = {'f^{max}'};    
+    end
+
+    properties
+        objective;
+        parameters = {1e-5};
+        epsilon = 1e-3; %
```
Comment (wahln) [inline optimization/+DoseConstraints/matRad_ObjectiveConstraint.m:29]: The epsilon is the slip factor? Maybe rename or comment it.
Does it make sense to have the f^max as a named parameter (in the parameter system) and the epsilon/slip not? shold both be just class properties / is there any scenario where using it from the GUI sensible?

File: optimization/+DoseObjectives/matRad_DoseObjective.m [hunk 4]
```diff
@@ -60,6 +60,10 @@
             s = struct@matRad_DoseOptimizationFunction(obj);
             s.penalty = obj.penalty;
         end
+        
+        function constr = convertToConstraint(obj,maxObj,epsilon)
```
Comment (wahln) [inline optimization/+DoseObjectives/matRad_DoseObjective.m:64]: See my comment about "obj" and "constr" before. Really name it "objective" here.

Comment (wahln) [reply optimization/+DoseObjectives/matRad_DoseObjective.m:64]: And why is this function in the objective? Couldn't I just pass the Objective to, for example, the constructor of the ObjectiveConstraint directly? Why do I need it?

File: optimization/+DoseObjectives/matRad_MaxDVH.m [hunk 5]
```diff
@@ -91,7 +91,7 @@
             deviation(dose < obj.parameters{1} | dose > d_ref2) = 0;
 
             % calculate delta
-            fDoseGrad = 2 * (obj.penalty/numel(dose))*deviation;
+            fDoseGrad = 2 * (1/numel(dose))*deviation;
```
Comment (wahln) [inline optimization/+DoseObjectives/matRad_MaxDVH.m:94]: 2/numel(dose) ?

**Review by wahln — CHANGES_REQUESTED** [review]
Looks good, just have some minor comments. Thanks!

**Pull request merged**